app/Modules/Bookings/BookingsService.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace App\Modules\Bookings;
+
+use App\Constants\ControllerRating;
+use App\Exceptions\OverlappingBookingException;
+use App\Position;
+use App\User;
+use Carbon\Carbon;
+
+class BookingsService implements BookingsServiceInterface
+{
+    /** @var mixed App\User */
+    protected $user;
+
+    public function __construct(User $user)
+    {
+        $this->user = $user;
+    }
+
+    /**
+     * Check if the the user is able to book the given position only on the rating requirement.
+     *
+     * @param  User  $user
+     * @param  Position  $position
+     * @return bool
+     */
+    public function validateRatingRequirement(User $user, Position $position): bool
+    {
+        $positionSuffix = $position->suffix;
+
+        $ratingValue = $user->atcRating->vatsim_id;
+
+        return ControllerRating::isValidRatingForSuffix($positionSuffix, $ratingValue);
+    }
+
+    public function validateBookingTimes(Carbon $from, Carbon $to, Position $position, int $excluded = null): bool
+    {
+        $bookings = Booking::where([['position_id', $position->getKey()], ['id', '!=', $excluded]]);
+
+        // Find between the times being booked for
+        $bookings->where(function ($query) use (&$from, &$to) {
+            // Where the start date is inside the booked time
+            $query->where(function ($query) use (&$from, &$to) {
+                $query->where('from', '>', $from)
+                    ->where('from', '<', $to);
+                // Or where the end date is inside the booked time
+            })->orWhere(function ($query) use (&$from, &$to) {
```

Comment:
Could simplify this using `whereBetween` - if either the start time or the end time is in the date range, it's a no-go.

Reply:
@atoff did you run into any specific problems if you did try this?

Reply:
Not quite sure why I wouldn't have, but done see a reason why we can't use it. Would have to keep the orWhere part of the second statement because a orWhereBetween doesn't exist afaik. But try it and the tests should let us know

Reply:
After chatting with @atoff, this isn't feasible as the whereBetween method using inclusive queries and this doesn't facilitate back-to-back bookings. /cc @AndyTWF